common/tools.py

```diff
@@ -369,6 +370,160 @@ def get_native_language_and_completeness(language_code):
+                        .format(path=path))
+        return False
+
+    # build full path
+    # <path>/backintime/<host>/<user>/<profile_id>
+    host, user, profile = cfg.hostUserProfile(profile_id)
+    full_path = path / 'backintime' / host / user / profile
+
+    # create full_path
+    try:
+        full_path.mkdir(mode=0o777, parents=True, exist_ok=True)
+
+    except PermissionError:
+        cfg.notifyError('\n'.join([
+            _('Creation of following folder failed:'),
+            str(full_path),
+            _(f'Write access may be restricted.')]))
+        return False
+
+    # Test filesystem
+    rc, msg = is_filesystem_valid(
+        full_path, path, mode, cfg.copyLinks())
+    if msg:
+        cfg.notifyError(msg)
+    if rc is False:
+        return False
+
+    # Test write access for the folder
+    rc, msg = is_writeable(full_path)
+    if msg:
+        cfg.notifyError(msg)
+    if rc is False:
+        return False
+
+    return True
+
+
+def is_filesystem_valid(full_path, msg_path, mode, copy_links):
+    """
+    Args:
+        full_path: The path to validate.
+        msg_path: The path used for display in error messages.
+        mode: Snapshot profile mode.
+        copy_links: Snapshot profiles copy links setting.
+
+    Returns:
+        (bool, str): A boolean value indicating success or failure and a
+            msg string.
+
+    """
+    fs = filesystem(full_path if isinstance(full_path, str) else str(full_path))
+
+    msg = None
+
+    # DEV NOTE
+    # notifyError() results in a messagebox
+    # Might be a candidate for a simple error-event-handler
+
+    if fs == 'vfat':
+        msg = _(
+            "Destination filesystem for {path} is formatted with FAT "
+            "which doesn't support hard-links. "
+            "Please use a native Linux filesystem.").format(path=msg_path)
+
+        return False, msg
+
+    elif fs == 'cifs' and not copy_links:
+        msg = _(
+            'Destination filesystem for {path} is an SMB-mounted share. '
+            'Please make sure the remote SMB server supports symlinks or '
+            'activate {copyLinks} in {expertOptions}.') \
+            .format(path=msg_path,
+                    copyLinks=_('Copy links (dereference symbolic links)'),
+                    expertOptions=_('Expert Options'))
+
+    elif fs == 'fuse.sshfs' and mode not in ('ssh', 'ssh_encfs'):
+        msg = _(
+            "Destination filesystem for {path} is an sshfs-mounted share."
+            " Sshfs doesn't support hard-links. "
+            "Please use mode 'SSH' instead.").format(path=msg_path)
+
+        return False, msg
+
+    return True, msg
+
+
+def is_writeable(folder):
+    """Test write access for the folder.
+
+    Args:
+        folder: The folder to check.
+
+    Returns:
+        (bool, str): A boolean value indicating success or failure and a
+            msg string.
+    """
+
+    if not isinstance(folder, pathlib.Path):
+        folder = pathlib.Path(folder)
```

Comment:
You can skip the check, as `Path(Path('example'))` will work. (Pathlib does its own `isinstance` check.)

```suggestion
    folder = pathlib.Path(folder)
```

Not sure which is strictly better. Depends on whether you're optimizing to minimise lines of code in BiT source, or instructions executed by Python.

The original code with the `isinstance` check is _slightly_ less work for Python, but this suggestion is less code for us.
See the pathlib initialisation source:
https://github.com/python/cpython/blob/77bb14622172c1a5d7d712bfe7050e7277d7ebb0/Lib/pathlib.py#L358